CIScripts/Common/Invoke-CommandWithFunctions.Tests.ps1

```diff
@@ -0,0 +1,134 @@
+Param (
+    [Parameter(Mandatory=$false)] [string] $TestenvConfFile,
+    [Parameter(ValueFromRemainingArguments=$true)] $UnusedParams
+)
+
+. $PSScriptRoot/Init.ps1
+. $PSScriptRoot/../Testenv/Testenv.ps1
+. $PSScriptRoot/../Testenv/Testbed.ps1
+
+. $PSScriptRoot/Invoke-CommandWithFunctions.ps1
+
+Describe "Invoke-CommandWithFunctions tests" -Tags CI, Systest {
+    function Test-CWF {
+        param(
+            [Parameter(Mandatory=$true)] [PSCustomObject] $FunctionsInvoked,
+            [Parameter(Mandatory=$true)] [ScriptBlock] $ScriptBlock,
+            [Switch] $CaptureOutput
+        )
+
+        Invoke-CommandWithFunctions -Session $Session `
+            -FunctionsInvoked $FunctionsInvoked `
+            -ScriptBlock $ScriptBlock `
+            -CaptureOutput:$CaptureOutput
+    }
+
+    function Test-SimpleFunction {
+        param(
+            [Parameter(Mandatory=$false)] [string] $SimpleParam = "A simple string to return"
+        )
+
+        return $SimpleParam
+    }
+
+    $TestSimpleFunctionInvoked = @(
```

Comment:
Maybe we could get rid of creating this structure every time, by 

1. passing just the function names to `Invoke-CommandWithFunctions`
2. Using `Get-Content function:$name` to get their bodies

?